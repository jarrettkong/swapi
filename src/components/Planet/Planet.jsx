import React, { Component } from 'react';
import PropTypes from 'prop-types';

class Person extends Component {
	state = {
		species: '',
		homeworld: {
			population: 0,
			name: ''
		}
	};
	componentDidMount() {
		this.fetchMissingData();
	}

	fetchHomeworld = homeworld => {
		return fetch(homeworld)
			.then(res => res.json())
			.then(planet => planet)
			.catch(err => console.log(err));
	};

	fetchSpecies = species => {
		return fetch(species)
			.then(res => res.json())
			.then(species => species)
			.catch(err => console.log(err));
	};

	fetchMissingData = () => {
		const speciesPromise = this.fetchSpecies(this.props.result.species[0]);
		const homeworldPromise = this.fetchHomeworld(this.props.result.homeworld);
		return Promise.all([speciesPromise, homeworldPromise]).then(data => {
			this.setState({
				species: data[0],
				homeworld: data[1]
			});
		});
	};

	render() {
		const { name } = this.props.result;
		const { species, homeworld } = this.state;
		return (
			<article className="Person">
				<h2>{name}</h2>
				<p>Species: {species.name}</p>
				<p>Homeworld: {homeworld.name}</p>
				<p>Homeworld Population: {homeworld.population}</p>
			</article>
		);
	}
}

Person.propTypes = {
	result: PropTypes.object.isRequired
};

export default Person;
