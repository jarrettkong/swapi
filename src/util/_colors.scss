$empire-blue: #445266;
$favorite-color: #d1000a;
$star-wars-yellow: #fed70f;
$main-color: #262626;
$background-color: #171315;
$light-grey: #7d7c77;
