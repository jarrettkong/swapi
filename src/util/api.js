export const fetchHomeworld = people => {
	const promises = people.map(person => {
		return fetch(person.homeworld)
			.then(res => res.json())
			.then(homeworld => ({ ...person, homeworld }));
	});
	return Promise.all(promises);
};

export const fetchResidents = planets => {
	const promises = planets.map(planet => {
		return fetchResidentNames(planet.residents).then(residents => ({ ...planet, residents }));
	});
	Promise.all(promises).then(planets => console.log(planets));
};

const fetchResidentNames = residents => {
	const promises = residents.map(resident => {
		return fetch(resident)
			.then(res => res.json())
			.then(data => data.name);
	});
	return Promise.all(promises);
};

export const fetchSpecies = people => {
	const promises = people.map(person => {
		return fetch(person.species[0])
			.then(res => res.json())
			.then(species => ({ ...person, species }));
	});
	return Promise.all(promises);
};

export const searchApi = term => {
	return fetch(`https://swapi.co/api/${term}`).then(response => {
		if (!response.ok) {
			throw new Error('Error adding grocery');
		}
		return response.json();
	});
};
