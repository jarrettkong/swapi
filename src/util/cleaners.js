export const cleanPeople = people => {
	return people.map(person => {
		return {
      'Name': person.name,
			'Species': person.species.name,
			'Homeworld': person.homeworld.name,
			'Population': person.homeworld.population,
			favorite: false
		};
	});
};

export const cleanPlanets = planets => {
	return planets.map(planet => {
		return {
      'Name': planet.name,
      'Population': planet.population,
      'Climate': planet.climate,
      'Terrain': planet.terrain,
      'Residents': planet.residents.join(', '),
			favorite: false
		};
	});
};

export const cleanVehicles = vehicles => {
	return vehicles.map(vehicle => {
		return {
      'Name': vehicle.name,
			'Model': vehicle.model,
			'Class': vehicle.class,
			'Passengers': vehicle.passengers,
			favorite: false
		};
	});
};
