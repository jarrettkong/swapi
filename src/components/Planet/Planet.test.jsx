import React from 'react';
import Planet from './Planet';
import { shallow } from 'enzyme';
import { mockPlanet } from '../../util/mockData';

describe('Planet', () => {
	let wrapper;
	beforeEach(() => {
		wrapper = shallow(<Planet result={mockPlanet} />);
	});

	it('should match the snapshot', () => {
		expect(wrapper).toMatchSnapshot();
	});
});
