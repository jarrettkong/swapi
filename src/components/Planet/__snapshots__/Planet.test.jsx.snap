// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Planet should match the snapshot 1`] = `ShallowWrapper {}`;
